src/org/zalando/stups/mint/worker/job/run.clj

```diff
@@ -22,17 +23,23 @@
   (try
     (when (time/after? (time/now)
                        @rate-limited-until)
-      (let [storage-url (config/require-config configuration :mint-storage-url)]
+      (let [storage-url (config/require-config configuration :mint-storage-url)
+            kio-url (config/require-config configuration :kio-url)]
         (log/debug "Starting new synchronisation run with %s..." configuration)
 
-        (let [apps (storage/list-apps storage-url tokens)]
-          (log/debug "Found apps: %s ..." apps)
-          (doseq [app apps]
+        (let [mint-apps (storage/list-apps storage-url tokens)
+              kio-apps (apps/list-apps kio-url tokens)]
+          (log/debug "Found apps: %s ..." mint-apps)
+          (doseq [mint-app mint-apps]
             (try
               (sync-app configuration
-                        app
+                        mint-app
+                        ; searching kio-app here
+                        (first (filter #(= (:id %)
```

Comment:
You know, we have THOUSANDS of apps, we need to SCALE :chart_with_upwards_trend: :smirk: